### ISSUE TITLE
[Bug/Support]: elpaca-try does not load package

### ISSUE THREAD

**Issue by ispringle** [progfolio/elpaca#253]
### Confirmation

- [X] I have checked the documentation (README, Wiki, docstrings, etc)
- [ ] I am  checking these without reading them.
- [X] I have searched previous issues to see if my question is a duplicate.

### Elpaca Version

Elpaca f19f4db HEAD
installer: 0.6
emacs-version 29.2 (build 1, aarch64-apple-darwin23.3.0, NS appkit-2487.40 Version 14.3)
git version 2.43.0

### Operating System

Sonoma 14.3 (M1)

### Description

When I `M-x elpaca-try` I get the prompt as I expect and can filter down to a single package. The package can be selected and it downloads/installs just like I expect from the output of Elpaca when a new package is added to my emacs config. However with elpaca-try that package that I wish to "try" never seems to get loaded. I can then call `load-file` and load the appropriate file to get the package working.

**Comment by progfolio** [comment]
Thanks for taking the time to fill out a ticket.
What package specifically?

**Comment by ispringle** [comment]
It doesn't seem to matter which package, however here are some I've tried (retried just now to verify):
- lispy
- lua-mode
- lsp-mode
- dracula-theme

**Comment by progfolio** [comment]
I'm unable to reproduce the issue on my end.
What do you get if you run the following test case?:


<details open><summary>Test Case</summary>

[How to run this test?](https://github.com/progfolio/elpaca/wiki/Troubleshooting#the-elpaca-test-macro)

```emacs-lisp
(elpaca-test
  :init (elpaca-try 'lispy)
  (elpaca-wait)
  (lispy-mode)
  (message "%S" minor-mode-list))
```

</details>
<details><summary>Host Env</summary>

<table>
<tr><td>elpaca</td><td>f19f4db HEAD -> master, origin/master, origin/HEAD</td>
<tr><td>installer</td><td>0.6</td>
<tr><td>emacs</td><td>GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.41, cairo version 1.18.0) of 2024-02-03</td>
<tr><td>git</td><td>git version 2.43.0</td>
</table>
</details>

<details><summary>Output</summary>

```emacs-lisp
Elpaca menu item cache discarded due to version change.
  INFO     Scraping files for loaddefs... 
  INFO     Scraping files for loaddefs...done
  GEN      ../elpaca-autoloads.el
Cloning into '/tmp/elpaca.5fzLnw/elpaca/repos/elpaca'...
Your branch is up to date with 'origin/master'.
Checking /tmp/elpaca.5fzLnw/elpaca/repos/elpaca...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/elpaca-info.el...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/elpaca-log.el...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/elpaca-manager.el...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/elpaca-menu-elpa.el...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/elpaca-menu-melpa.el...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/elpaca-menu-org.el...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/elpaca-process.el...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/elpaca-test.el...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/elpaca-ui.el...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/elpaca.el...
Checking /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/doc...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/doc/early-init.el...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/doc/init.el...
Checking /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/extensions...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/extensions/elpaca-use-package.el...
Checking /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/images...
Checking /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/test...
Compiling /tmp/elpaca.5fzLnw/elpaca/repos/elpaca/test/elpaca-test.el...
Done (Total of 11 files compiled, 3 skipped in 4 directories)

Downloading MELPA recipes...
Downloading MELPA recipes...100%
Downloading NonGNU-devel ELPA...
Downloading GNU-devel ELPA...
Downloading NonGNU ELPA...
Downloading GNU ELPA...
(lispy-other-mode lispy-goto-mode lispy-mode delete-selection-mode avy-linum-mode xref-etags-mode edebug-backtrace-mode edebug-mode org-cdlatex-mode org-src-mode comint-fontify-input-mode org-list-checkbox-radio-mode orgtbl-mode org-table-follow-field-mode org-table-header-line-mode outline-minor-mode global-hl-line-mode hl-line-mode header-line-indent-mode global-display-line-numbers-mode display-line-numbers-mode elpaca-ui-tail-mode elpaca-ui-live-update-mode elpaca-no-symlink-mode cl-old-struct-compat-mode tooltip-mode global-eldoc-mode eldoc-mode show-paren-mode electric-quote-mode electric-layout-mode electric-indent-mode mouse-wheel-mode modifier-bar-mode tool-bar-mode paragraph-indent-minor-mode global-prettify-symbols-mode prettify-symbols-mode use-hard-newlines tab-bar-history-mode tab-bar-mode menu-bar-mode file-name-shadow-mode isearch-fold-quotes-mode lost-selection-mode horizontal-scroll-bar-mode context-menu-mode jit-lock-debug-mode global-font-lock-mode font-lock-mode blink-cursor-mode window-divider-mode undelete-frame-mode minibuffer-regexp-mode completion-in-region-mode visible-mode buffer-read-only size-indication-mode column-number-mode line-number-mode auto-fill-function global-visual-line-mode visual-line-mode indent-tabs-mode transient-mark-mode cursor-face-highlight-mode read-extended-command-mode next-error-follow-minor-mode auto-composition-mode auto-encryption-mode auto-compression-mode temp-buffer-resize-mode button-mode lock-file-mode auto-save-visited-mode auto-save-mode auto-fill-mode abbrev-mode overwrite-mode view-mode hs-minor-mode)

 Test Env

Elpaca f19f4db HEAD -> master, origin/master, origin/HEAD
installer:      0.6
emacs-version:  GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.41, cairo version 1.18.0)
 of 2024-02-03
git --version:  git version 2.43.0
```
</details>

**Comment by ispringle** [comment]
Well that's failing to run, getting an eager macro expansion error. However on the plus side, while debugging why elpaca-test isn't working I discovered the culprit and resolved my issue. I had previously been trying out Nix for package management, which involves using their emacs package, however this doesn't have an emacs-version, so I had to manually set the `elpaca-core-date`. Seems if this is wrong, then things stop working, so removing that line resolved the elpaca-try not loading.

Should be good to close this I guess, it seems reasonable that the wrong emacs version date would mess stuff up I think.
